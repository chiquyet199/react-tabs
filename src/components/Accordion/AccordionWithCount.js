import React from "react";
import {number, string} from "prop-types";
import AccordionWithCountTitle from "./AccordionWithCountTitle";
import {Accordion} from "./Accordion";

export const AccordionWithCount = props => (
  <Accordion
    title={({collapsible}) => (
      <AccordionWithCountTitle
        {...{
          ...props,
          collapsible,
        }}
      />
    )}
  >
    {props.children}
  </Accordion>
);

AccordionWithCount.propTypes = {
  headerIconName: string.isRequired,
  headerCount: number.isRequired,
  title: string.isRequired,
};

export default AccordionWithCount;
