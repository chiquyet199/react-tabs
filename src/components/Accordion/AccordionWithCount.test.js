import React from "react";
import {Platform, Text, View} from "react-native";
import {shallow} from "enzyme";
import {AccordionWithCount} from "./AccordionWithCount";

describe("AccordionWithCount", () => {
  it(`matches snapshot on ${Platform.OS}`, () => {
    expect(getComponent()).toMatchSnapshot();
  });

  const getComponent = props => {
    const parsedProps = {
      children: (
        <View>
          <Text>Lorem ipsum</Text>
        </View>
      ),
      headerIconName: "rentpassport",
      headerCount: 99,
      title: "title",
      ...props,
    };

    return shallow(<AccordionWithCount {...parsedProps} />);
  };
});
