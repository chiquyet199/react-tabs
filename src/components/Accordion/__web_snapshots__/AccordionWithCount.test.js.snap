// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionWithCount matches snapshot on web 1`] = `
<Accordion
  title={[Function]}
>
  <View>
    <Text>
      Lorem ipsum
    </Text>
  </View>
</Accordion>
`;
